react-hot-button/my-app/src/HotButton.tsx

```diff
@@ -0,0 +1,28 @@
+import { useState } from 'react';
+import './HotButton.css';
+
+export function HotButton() {
+  const [count, setCount] = useState(0);
+  let colorClass: string = 'firstColor';
+  if (count <= 3) {
+    colorClass = 'firstColor';
+  } else if (count <= 6) {
+    colorClass = 'secondColor';
+  } else if (count <= 9) {
+    colorClass = 'thirdColor';
+  } else if (count <= 12) {
+    colorClass = 'fourthColor';
+  } else if (count <= 15) {
+    colorClass = 'fifthColor';
+  } else if (count <= 18) {
```

Comment:
If the button is clicked more than 18 times it will reset to the 'cold' color. 
The last assignment could be in an else block instead of an else if.